src/bits.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use error;
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct BitLength {
```

Comment:
I agree it's weird to have `bits::BitLength` but I also think `bits::Length` or `bit::Length` is too clever. I found out that these names that only make sense with module prefixes don't work well in rustdoc (see `error::Unspecified` which rustdoc presents as just `Unspecified`, which is confusing).

I think we probably should create a "prelude" of common types that are used internally in *ring* and use those types (only) unprefixed. I think `BitLength` is a prime candidate for that.